pkg/odo/cli/component/create_helpers.go

```diff
@@ -91,3 +92,16 @@ func (co *CreateOptions) DevfileJSON() error {
 	machineoutput.OutputSuccess(cfd.GetComponent())
 	return nil
 }
+
+// GetComponentTypeFromDevfile returns component type from the devfile metadata
+func GetComponentTypeFromDevfile(metadata devfile.DevfileMetadata) string {
+	var componentType string
+	if metadata.ProjectType != "" {
+		componentType = metadata.ProjectType
+	} else if metadata.Language != "" {
+		componentType = metadata.Language
+	} else {
+		componentType = metadata.Name
```

Comment:
This is causing the problem mentioned in https://github.com/openshift/odo/pull/4763#issuecomment-862246661 IIUC.

Reply:
I think `metadata.Name` is not reliable to know component type because if user uses a custom name, odo will pick it up as component type. So if `metadata.Language` is not available (for scneario like https://github.com/openshift/odo/issues/4816), we should set `componentType` to some sensible default like `Not Available` or somethin else.